elytron/src/main/java/org/wildfly/extension/elytron/MapperParser.java

```diff
@@ -17,1986 +17,216 @@
-        }
-
-        if (requiredAttributes.isEmpty() == false) {
-            throw missingRequired(reader, requiredAttributes);
-        }
-
-        requireNoContent(reader);
-
-        addRoleMapper.get(OP_ADDR).set(parentAddress).add(LOGICAL_ROLE_MAPPER, name);
-        operations.add(addRoleMapper);
-    }
-
-    private void startMappers(boolean started, XMLExtendedStreamWriter writer) throws XMLStreamException {
-        if (started == false) {
-            writer.writeStartElement(MAPPERS);
-        }
-    }
-
-    private boolean writeCustomPermissionMappers(boolean started, ModelNode subsystem, XMLExtendedStreamWriter writer) throws XMLStreamException {
-        if (subsystem.hasDefined(CUSTOM_PERMISSION_MAPPER)) {
-            startMappers(started, writer);
-            ModelNode principalDecoders = subsystem.require(CUSTOM_PERMISSION_MAPPER);
-            for (String name : principalDecoders.keys()) {
-                ModelNode principalDecoder = principalDecoders.require(name);
-
-                writeCustomComponent(CUSTOM_PERMISSION_MAPPER, name, principalDecoder, writer);
-            }
-
-            return true;
-        }
-
-        return false;
-    }
-
-    private boolean writeLogicalPermissionMappers(boolean started, ModelNode subsystem, XMLExtendedStreamWriter writer) throws XMLStreamException {
-        if (subsystem.hasDefined(LOGICAL_PERMISSION_MAPPER)) {
-            startMappers(started, writer);
-            ModelNode permissionMappers = subsystem.require(LOGICAL_PERMISSION_MAPPER);
-            for (String name : permissionMappers.keys()) {
-                ModelNode permissionMapper = permissionMappers.require(name);
-                writer.writeStartElement(LOGICAL_PERMISSION_MAPPER);
-                writer.writeAttribute(NAME, name);
-                PermissionMapperDefinitions.LOGICAL_OPERATION.marshallAsAttribute(permissionMapper, writer);
-                PermissionMapperDefinitions.LEFT.marshallAsAttribute(permissionMapper, writer);
-                PermissionMapperDefinitions.RIGHT.marshallAsAttribute(permissionMapper, writer);
-                writer.writeEndElement();
-            }
-
-            return true;
-        }
-
-        return false;
-    }
-
-    private boolean writeSimplePermissionMappers(boolean started, ModelNode subsystem, XMLExtendedStreamWriter writer) throws XMLStreamException {
-        if (subsystem.hasDefined(SIMPLE_PERMISSION_MAPPER)) {
-            startMappers(started, writer);
-            ModelNode permissionMappers = subsystem.require(SIMPLE_PERMISSION_MAPPER);
-            for (String name : permissionMappers.keys()) {
-                ModelNode permissionMapper = permissionMappers.require(name);
-                writer.writeStartElement(SIMPLE_PERMISSION_MAPPER);
-                writer.writeAttribute(NAME, name);
-                PermissionMapperDefinitions.MAPPING_MODE.marshallAsAttribute(permissionMapper, false, writer);
-                if (permissionMap
```

Comment:
One small comment on this one, legacy becomes relative after a while - to make it clear wouldn't t be better to say version1Format or something like that?

Reply:
Yeah I see this is just to represent the different simple permission mapper, I think in this case we would be better to just take a version is a parameter so the correct version can be returned based on namespace.

Reply:
yeah, naming variables is always the hardest :) will update it.

Reply:
fixed